File: files/CH-095.py
import pandas as pd

path = "CH-095 Last Inventory.xlsx"
input = pd.read_excel(path, usecols = "B:G", skiprows = 1)
test  = pd.read_excel(path, usecols = "i:j", skiprows = 1)
test.columns = test.columns.str.replace(".1", "")

result = input.copy()
result['Last Inventory'] = input.iloc[:, 1:]\
    .apply(lambda x: x[x.last_valid_index()], axis = 1)\
    .astype("int64")
result = result[['Product', 'Last Inventory']]

print(result.equals(test)) # True

File: files/CH-095.R
library(tidyverse)
library(readxl)

path = "files/CH-095 Last Inventory.xlsx"
input = read_excel(path, range = "B2:G7")
test  = read_excel(path, range = "I2:J7")


result = input %>%
  rowwise() %>%
  mutate(`Last Inventory` = last(na.omit(c_across(-Product)))) %>%
  ungroup() %>%
  select(c(1,7))

identical(result, test)         
# [1] TRUE